From 686bc470ce39630c37fd7e3436ba3bc706a8f040 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Tue, 25 Oct 2016 19:34:24 +0100
Subject: [PATCH 1/6] VirtSubproc: open arbitrary files in binary mode

It doesn't actually matter when we only use their fileno(), but it's
better to be consistent.

Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 lib/VirtSubproc.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/VirtSubproc.py b/lib/VirtSubproc.py
index ba87740..1aeb673 100644
--- a/lib/VirtSubproc.py
+++ b/lib/VirtSubproc.py
@@ -38,7 +38,7 @@ import shutil
 import adtlog
 
 progname = "<VirtSubproc>"
-devnull_read = open('/dev/null', 'r')
+devnull_read = open('/dev/null', 'rb')
 caller = __main__
 copy_timeout = int(os.getenv('AUTOPKGTEST_VIRT_COPY_TIMEOUT', '300'))
 
@@ -512,9 +512,9 @@ def copyupdown_internal(wh, sd, upp):
     if not dirsp:
         rune = 'cat %s%s' % ('><'[upp], remfileq)
         if upp:
-            deststdout = open(sd[idst], 'w')
+            deststdout = open(sd[idst], 'wb')
         else:
-            srcstdin = open(sd[isrc], 'r')
+            srcstdin = open(sd[isrc], 'rb')
             status = os.fstat(srcstdin.fileno())
             if status.st_mode & 0o111:
                 rune += '; chmod +x -- %s' % (remfileq)
-- 
2.10.1

